send_email_pdf/main.py

```diff
@@ -0,0 +1,293 @@
+                       'line': {"color": "white",
+                                "width": 3}},
+               'bgcolor': "white"},
+        title={'text': "Daily Mail Average Sentiment"}))
+    daily_mail_fig.write_image("daily_mail_plot.svg")
+
+    template = f'''
+<html>
+<head>
+<style>
+    /* Define the CSS styles for your dashboard here */
+
+    .title-container {{
+        display: flex;
+        flex-direction: row;
+    }}
+
+    
+    .widget {{
+        background-color: #fff;
+        padding: 9px;
+        margin-bottom: 10px;
+        border-radius: 5px;
+        box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
+    }}
+
+    .sentiment-container {{
+        display: flex;
+        justify-content: center;
+    }}
+    /* Add more styles as needed */
+</style>
+</head>
+<body>
+<div class="title-container">
+    <img style="width: 50px; height: 50px; margin-bottom: 10px;" src="SL_Favicon-45.png" alt="Logo" class=title-container/>
+    <span style="text-align:center;font-size:250%;">Media Sentiment Daily Quarter Report</span>
+</div>
+
+<div class="widget">
+    <div class="sentiment-container">
+        <img style="width: 300px; height: 200px" src = "bbc_plot.svg" alt="BBC"/>
+        <img style="width: 300px; height: 200px" src = "daily_mail_plot.svg" alt="Daily Mail"/>
+    </div>
+</div>
+
+<div class="widget">
+    <h1>Highest article sentiment stories</h1>
+    {get_titles(top_5_stories_titles)}
+</div>
+
+<div class="widget">
+    <h1>Lowest article sentiment stories</h1>
+    {get_titles(lowest_5_stories_titles)}
+</div>
+
+<div class="widget">
+    <h1>Highest reddit sentiment stories</h1>
+    {get_titles(top_5_reddit_titles)}
+</div>
+
+<div class="widget">
+    <h1>Lowest reddit sentiment stories</h1>
+    {get_titles(lowest_5_reddit_titles)}
+</div>
+
+<!-- Add more widgets as needed -->
+</body>
+</html>
+'''
+    return template
+
+
+def convert_html_to_pdf(html_template: str) -> None:
+    """Converts the HTML template provided into a pdf report file"""
+    # open output file for writing (truncated binary)
+    if not isinstance(html_template, str):
+        raise ValueError("The HTML template should be provided as a string")
+    if not isinstance(PDF_FILE_NAME, str):
+        raise ValueError("The file name should be provided as a string")
+    result_file = open(PDF_FILE_NAME, "w+b")
+
+    # convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        html_template,                # the HTML to convert
+        dest=result_file)           # file handle to receive result
+
+    # close output file
+    result_file.close()
+
+    # return True on success and False on errors
+    return pisa_status.err
+
+
+def upload_to_s3() -> None:
+    """Function that uploads the created pdf to an s3 bucket"""
+    print("Establishing connection to AWS.")
+    s3_client = client("s3", aws_access_key_id=environ.get("ACCESS_KEY"),
+                       aws_secret_access_key=environ.get("SECRET_KEY"))
```

Comment:
Coding style nit pick:
Would make it easier to test if S3 object is passed in as a parameter and lines 226-227 are within handler, but doesn't matter too much since S3 is only used once